conjure-java-core/src/main/java/com/palantir/conjure/java/util/Goethe.java

```diff
@@ -40,19 +44,44 @@ private Goethe() {}
 
     /** Formats the given Java file and emits it to the appropriate directory under {@code baseDir}. */
     public static Path formatAndEmit(JavaFile file, Path baseDir) {
+        StringBuilder code = new StringBuilder();
         try {
             Path outputFile = getFilePath(file, baseDir);
-            StringBuilder code = new StringBuilder();
             file.writeTo(code);
 
             CharSink sink = com.google.common.io.Files.asCharSink(outputFile.toFile(), StandardCharsets.UTF_8);
             JAVA_FORMATTER.formatSource(CharSource.wrap(code), sink);
             return outputFile;
-        } catch (IOException | FormatterException e) {
+        } catch (FormatterException e) {
+            throw new RuntimeException(generateMessage(file, code.toString(), e.diagnostics()), e);
+        } catch (IOException e) {
             throw new RuntimeException(e);
         }
     }
 
+    private static String generateMessage(
+            JavaFile file, String unformattedSource, List<FormatterDiagnostic> formatterDiagnostics) {
+        try {
+            List<String> lines = Splitter.on('\n').splitToList(unformattedSource);
+            StringBuilder failureText = new StringBuilder();
+            failureText.append("Failed to format '")
+                    .append(file.packageName).append('.').append(file.typeSpec.name).append("'\n");
+            for (FormatterDiagnostic formatterDiagnostic : formatterDiagnostics) {
+                failureText.append(formatterDiagnostic.message()).append("\n")
+                        // Diagnostic values are one-indexed, while our list is zero-indexed.
+                        .append(lines.get(formatterDiagnostic.line() - 1))
+                        .append('\n')
+                        // Offset by two to convert from one-indexed to zero indexed values, and account for the
+                        // carrot.
+                        .append(Strings.repeat(" ", Math.max(0, formatterDiagnostic.column() - 2)))
+                        .append("^\n\n");
```

Comment:
this seems fine but i'm surprised they don't provide this kind of diagnostic printer

Reply:
Me too, there are no code references to the line/column values, only printing diagnostic error message strings.